std/stdio.d

```diff
@@ -498,17 +498,17 @@ struct File
     private Impl* _p;
     private string _name;
 
-    package this(FILE* handle, string name, uint refs = 1, bool isPopened = false) @trusted
+    package this(FILE* handle, string name, uint refs = 1, bool isPopened = false) @trusted //@nogc nothrow
     {
         import core.stdc.stdlib : malloc;
         import std.exception : enforce;
 
         assert(!_p);
-        _p = cast(Impl*) enforce(malloc(Impl.sizeof), "Out of memory");
+        _p = cast(Impl*) enforce(malloc(Impl.sizeof), preAllocatedStdioOOMError);
```

Comment:
```suggestion
        _p = cast(Impl*) malloc(Impl.sizeof);
        if (!p)
        {
            import core.exception : onOutOfMemoryError;
            onOutOfMemoryError();
        }
```

Reply:
unfortunately i saw this suggestion immediately after implementing it word for word on my own lol

Reply:
actually, i think yours might be clearer to understand than my version. what do you think?